Changelog.md

```diff
@@ -1,3 +1,10 @@
+## Unreleased
+
+### Bug Fixes
+
+* Fix a bug that caused `findIndices` to increment the index during
+  a Skip condition.
```

Comment:
Just write user facing behavior in the changelog, not internal details like `Skip`. The goal is to explain in sucha way that the user of the library can understand the bug behavior and identify it when they hit it without having to understand the design details of the library.